examples/first.cpp

```diff
@@ -24,27 +27,85 @@ std::string gencode(Context& ctx)
             maxval = x;
             maxpos = i;
         ctx.endif_();
-        i += ctx.const_((int)sizeof(int));
+        i += elemsize;
     ctx.while_(i < n);
+    minpos /= elemsize;
+    maxpos /= elemsize;
     store_<int>(minpos_addr, minpos);
     store_<int>(maxpos_addr, maxpos);
     ctx.return_(ctx.const_(0));
     ctx.endfunc();
-    //return (minmaxfunc_t)ctx.getfunc("foo").ptr();
-    return *((std::string*)(ctx.getfunc("foo").ptr()));
-}
+    return ctx.getfunc("foo");
 }
 
+//typedef int (*justcompiled_t)(const int* ptr, int64_t n, int* out);
+//void* gencode(loops::Context& ctx)
+//{
+//    using namespace loops;
+//    IReg ptr, n, out;
+//    ctx.startfunc("foo", {&ptr, &n, &out});
+//    IReg i = ctx.const_(0);
+//    IReg sum = ctx.const_(0);
+//    IReg elemsize = ctx.const_((int)sizeof(int));
+//    n *= elemsize;
+//    ctx.do_();
+//        IReg x = load_<int>(ptr, i);
+//        sum += x;
+//        i += elemsize;
+//    ctx.while_(i < n);
+//    ctx.return_(sum);
+//    ctx.endfunc();
+//    void* res = ctx.getfunc("foo").ptr();
+//    return res;
+//}
+
+//typedef int (*justcompiled_t)(int64_t a, int64_t b);
+//void* gencode(loops::Context& ctx)
+//{
+//    using namespace loops;
+//    IReg a, b;
+//    ctx.startfunc("foo", {&a, &b});
+//    ctx.if_( a > b );
+//    b = a;
+//    ctx.endif_();
+//    ctx.return_(b);
+//    ctx.endfunc();
+//    void* res = ctx.getfunc("foo").ptr();
+//    return res;
+//}
+
 int main(int argc, char** argv)
 {
-    loops::Context ctx(loops::Compiler::make_virtual_dump());
-    std::string f = gencode(ctx);
-    std::cout << f << std::endl;
-    //minmaxfunc_t f = gencode(ctx);
-    //std::vector<int> v = { 1, 2, -5, 7, 8 };
-    //int minpos = -1, maxpos = -1;
-    //int retval = f(&v[0], (int64_t)v.size(), &minpos, &maxpos);
-    //printf("retval=%d, minpos=%d, maxpos=%d\n", retval, (int)minpos, (int)maxpos);
+    {//Function usage
+        loops::Context ctx(loops::Backend::make_aarch64_compiler());
```

Comment:
since this is JIT, i.e. the compiled code is supposed to be immediately run on the same machine, the default constructor should automatically use backend for the current platform